website/docs/r/web_pubsub_hub.html.markdown

```diff
@@ -0,0 +1,126 @@
+---
+subcategory: "Web Publishing Subscription"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_web_pubsub_hub"
+description: |-
+  Manages the hub settings for a Web Pubsub service.
+---
+
+# azurerm_web_pubsub_hub
+
+Manages the hub settings for a Web Pubsub.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "terraform-webpubsub"
+  location = "east us"
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  name                = "tfex-uai"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_web_pubsub" "example" {
+  name                = "tfex-webpubsub"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  sku      = "Standard_S1"
+  capacity = 1
+}
+
+resource "azurerm_web_pubsub_hub" "test" {
+  name          = "tfex-wpsh"
+  web_pubsub_id = azurerm_web_pubsub.exmaple.id
+  event_handler {
+
+    setting {
+      url_template       = "https://test.com/api/{hub}/{event}"
+      user_event_pattern = "*"
+      system_events      = ["connect", "connected"]
+    }
+
+    setting {
+      url_template       = "https://test.com/api/{hub}/{event}"
+      user_event_pattern = "event1, event2"
+      system_events      = ["connected"]
+      auth {
+        managed_identity_id = azurerm_user_assigned_identity.test.id
+      }
+    }
+  }
+  anonymous_connections_enabled = true
+
+  depends_on = [
+    azurerm_web_pubsub.test
+  ]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Web Pubsub hub service. Changing this forces a new resource to be created.
+
+* `web_pubsub_id` - (Required) Specify the id of the Web Pubsub. Changing this forces a new resource to be created.
+
+* `event_handler` - (Required) An `event_handler` block as defined below.
+
+* `anonymous_connections_enabled` - (Optional) Is anonymous connections are allowed for this hub? Defaults to `false`. Possible value are `true`, `false`
+
+---
+
+An `event_handler` block supports the following:
+
+* `setting` - (Required) An `setting` block as defined below.
```

Comment:
why is this two blocks deep? would it not make sense to remove the settings block and make event_handler a list of the settings blocks at the top level?

Reply:
Thanks for the suggestion, I changed the schema definition accordingly